pallets/staking/src/lib.rs

```diff
@@ -487,20 +489,32 @@ pub struct ValidatorPrefs {
     pub commission: Perbill,
 }
 
-// TODO: Remove before mainnet - only used for storage upgrade
-/// Commission can be set globally or by validator
+/// Preference of an identity regarding validation.
 #[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-pub enum Commission {
-    /// Flag that allow every validator to have individual commission.
-    Individual,
-    /// Every validator has same commission that set globally.
-    Global(Perbill),
+pub struct PermissionedIdentityPrefs {
+    /// Intended no. of validator counts an identity wants to run.
+    /// It act as an hard limit on no. of validators an identity can run but it can be
+    /// amended using governance.
+    ///
+    /// It should satisfy count < MaxValidatorPerIdentity * Self::validator_count().
+    pub intended_count: u32,
+    /// Keep track of running no. of validators of an DID.
+    pub running_count: u32,
 }
 
-impl Default for Commission {
+impl Default for PermissionedIdentityPrefs {
     fn default() -> Self {
-        Commission::Individual
+        // By default only 1 validator is allowed to run by an identity.
+        Self::new(1)
+    }
+}
+
+impl PermissionedIdentityPrefs {
+    fn new(count: u32) -> Self {
```

Comment:
This constructor is not public, Do we need it ?

Reply:
Hmm, I think we can do it for future reference.